source/extensions/filters/network/thrift_proxy/router/upstream_request.cc

```diff
@@ -302,20 +302,21 @@ bool UpstreamRequest::onResetStream(ConnectionPool::PoolFailureReason reason) {
   bool close_downstream = true;
 
   chargeResponseTiming();
-
+  LocalExceptionErrorType error_code = LocalExceptionErrorType::Overflow;
   switch (reason) {
   case ConnectionPool::PoolFailureReason::Overflow:
-    stats_.incResponseLocalException(parent_.cluster());
+    stats_.incResponseLocalException(parent_.cluster(), error_code);
     parent_.sendLocalReply(AppException(AppExceptionType::InternalError,
                                         "thrift upstream request: too many connections"),
                            false /* Don't close the downstream connection. */);
     close_downstream = false;
     break;
   case ConnectionPool::PoolFailureReason::LocalConnectionFailure:
-    FALLTHRU;
+    error_code = LocalExceptionErrorType::LocalConnectionFailure;
   case ConnectionPool::PoolFailureReason::RemoteConnectionFailure:
-    FALLTHRU;
+    error_code = LocalExceptionErrorType::RemoteConnectionFailure;
   case ConnectionPool::PoolFailureReason::Timeout:
+    error_code = LocalExceptionErrorType::Timeout;
```

Comment:
This fails CI because the local/remote connection failure cases fall through with a `FALLTHRU` marker.

But, even with the markers this should fail unit tests because the error code will always be Timeout.